src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,43 +1,49 @@
 import React from 'react';
-import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+
+type Props = {
+  todo: Todo;
+  user: User;
+  onClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, user, onClose }) => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
-        <Loader />
-      ) : (
         <div className="modal-card">
           <header className="modal-card-head">
             <div
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {todo.id}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
+            <button type="button" className="delete" data-cy="modal-close" onClick={onClose}/>
           </header>
 
           <div className="modal-card-body">
             <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
+              {todo.title}
             </p>
 
             <p className="block" data-cy="modal-user">
               {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
+              <strong className="has-text-danger">{todo.completed ? 'Completed' : 'Active'}</strong>
 
               {' by '}
 
-              <a href="mailto:Sincere@april.biz">Leanne Graham</a>
+              <a href={`mailto:${user?.email}`}>{user?.name}</a>
```

Comment:
The use of optional chaining (`user?.email` and `user?.name`) is unnecessary here because `user` is a required prop according to the `Props` type definition. You can safely access `user.email` and `user.name` directly.